File: leetcode/math/2927_Distribute_Candies_Among_Children_III/codes/comb.c
#include <stdio.h>

long long C2(long long x) {
    return x >= 2 ? x * (x - 1) / 2 : 0;
}

long long distributeCandies(int n, int limit) {
    long long total = C2(n + 2);
    long long sub1 = 3 * C2(n - (limit + 1) + 2);
    long long add2 = 3 * C2(n - 2 * (limit + 1) + 2);
    long long sub3 = C2(n - 3 * (limit + 1) + 2);
    return total - sub1 + add2 - sub3;
}

File: leetcode/math/2927_Distribute_Candies_Among_Children_III/codes/comb.py
class Solution:
    def distributeCandies(self, n: int, limit: int) -> int:
        def C2(x):
            return (x * (x - 1)) // 2 if x >= 2 else 0

        total = C2(n + 2)
        sub1 = 3 * C2(n - (limit + 1) + 2)
        add2 = 3 * C2(n - 2 * (limit + 1) + 2)
        sub3 = C2(n - 3 * (limit + 1) + 2)

        return total - sub1 + add2 - sub3
